src/datajudge/utils.py

```diff
@@ -42,3 +42,138 @@ def format_difference(
+
+    If the first element is not instanceof tuple,
+    each element will be transparently packaged into a 1-tuple for processing;
+    this process is not visible to the caller.
+
+    Handles None values as described in `sort_tuple_none_aware`.
+    """
+    collection = list(collection)
+    if not isinstance(collection[0], tuple):
+        # package into a 1 tuple and pass into the method again
+        packaged_list = [(elem,) for elem in collection]
+        res_main, res_counts = output_processor_sort(packaged_list, counts)
+        return [elem[0] for elem in res_main], res_counts
+
+    if counts is None:
+        return sort_tuple_none_aware(collection), counts
+
+    assert len(collection) == len(
+        counts
+    ), "collection and counts must have the same length"
+
+    if len(collection) <= 1:
+        return collection, counts  # empty or 1 element lists are always sorted
+
+    lst = sort_tuple_none_aware(
+        [(-count, *elem) for count, elem in zip(counts, collection)]
+    )
+    return [elem[1:] for elem in lst], [-elem[0] for elem in lst]
+
+
+def output_processor_limit(
+    collection: Collection, counts: Optional[Collection] = None, limit: int = 100
+) -> Collection:
+    """
+    Limits the collection to the first ``limit`` elements.
+    If the list was shortened,
+    will add a ``limit+1``-th string element,
+    informing the user of the truncation.
+    The default limit of ``100`` can be adjusted using ``functools.partial``
+    """
+    collection = list(collection)
+
+    ret_collection = collection[:limit]
+    ret_counts = None if counts is None else list(counts)[:limit]
+    if len(collection) > limit:
+        ret_collection.append(
+            f"<SHORTENED OUTPUT, displaying the first {limit} / {len(collection)} elements above>"
+        )
+        if ret_counts is not None:
+            ret_counts.append(
+                f"<SHORTENED OUTPUT, displaying the first {limit} / {len(collection)} counts above>"
+            )
+
+    return ret_collection, ret_counts
+
+
+def filternull_element(values: List) -> List:
+    return [value for value in values if value is not None]
+
+
+def filternull_never(values: List) -> List:
+    return values
+
+
+def filternull_element_or_tuple_all(values: List) -> List:
+    return [
+        value
+        for value in values
+        if value is not None
+        and not (isinstance(value, tuple) and all(x is None for x in value))
+    ]
+
+
+def filternull_element_or_tuple_any(values: List) -> List:
+    return [
+        value
+        for value in values
+        if value is not None
+        and not (isinstance(value, tuple) and any(x is None for x in value))
+    ]
+
+
+def sort_tuple_none_aware(collection: Collection[Tuple], ascending=True):
+    """
+    Sorts a collection of either tuples or single elements,
+    where `None` is considered the same as the default value of the respective column's type.
```

Comment:
```suggestion
    where ``None`` is considered the same as the default value of the respective column's type.
```

What column are we referring to, here? Looking at this function in isolation it might not be too clear.

Reply:
Updated with more detailed explanation